common/stream.h

```diff
@@ -525,13 +557,13 @@ inline stream_out_t::stream_out_t()
 template<typename TType>
 inline void stream_out_t::push(const TType& value)
 {
-	if constexpr (std::is_trivial_v<TType> &&
+	if constexpr (std::is_trivially_copyable_v<TType> &&
 	              !(std::is_pointer_v<TType> ||
 	                std::is_reference_v<TType>))
 	{
 		integer_t size = outBuffer.size();
 		outBuffer.resize(size + sizeof(TType));
-		memcpy(&outBuffer[size], &value, sizeof(TType));
+		reinterpret_cast<TType&>(outBuffer[size]) = value;
```

Comment:
Same thing here, IMHO
```cpp
if constexpr (std::is_trivially_copyable_v<TType>)
{
    auto data = reinterpret_cast<const uint8_t*>(&value);
    outBuffer.insert(outBuffer.end(), data, data + sizeof(TType));
}
```
is much more cleaner

Reply:
```c++
if constexpr (std::is_trivially_copyable_v<TType>)
{
	auto& data = reinterpret_cast<const uint8_t(&)[sizeof(TType)]>(value);
	outBuffer.insert(outBuffer.end(), std::begin(data), std::end(data));
}
```
IMHO is even cleaner